homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,37 @@
+#include <algorithm>
+#include <iostream>
+#include "compression.hpp"
+
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& image_array) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width * height);
+
+    std::for_each(image_array.begin(), image_array.end(),[&compressed](auto& row){
```

Comment:
for_each for me is similar to for. You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )